kieker-extension-cassandra/src/kieker/extension/cassandra/reader/CassandraDb.java

```diff
@@ -0,0 +1,164 @@
+/***************************************************************************
+ * Copyright 2018 Armin Moebius, Sven Ulrich (https://www.rbee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+package kieker.extension.cassandra.reader;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.exceptions.AuthenticationException;
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import com.datastax.driver.core.exceptions.QueryExecutionException;
+import com.datastax.driver.core.exceptions.QueryValidationException;
+import com.datastax.driver.core.exceptions.UnsupportedFeatureException;
+import com.datastax.driver.core.policies.DowngradingConsistencyRetryPolicy;
+
+import kieker.common.exception.ConfigurationException;
+
+/**
+ *
+ * @author Armin Moebius, Sven Ulrich
+ *
+ */
+public class CassandraDb {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(CassandraDb.class);
+
+	private Cluster cluster;
+	private final List<InetSocketAddress> contactPoints;
+	private final String keyspace;
+	private Session session;
```

Comment:
Shouldn't this be autoclosable and close the session on finishing?

Reply:
I tried that, but it seems not to work. This will need further testing. Also it should be ported to Cassandra's native library, but I would rather do this in a separate step and ticket.